### PR TITLE
Add Data Mate date functions (Part 1)

### PR DESCRIPTION
**Pull request by peterdemartini**
For a list of functions to add see https://github.com/terascope/teraslice/issues/2695

### CODE REVIEW

File: packages/data-mate/src/function-configs/string/trim.ts [hunk 1]
```diff
@@ -105,10 +112,5 @@ export const trimConfig: FieldTransformConfig<TrimArgs> = {
             array: false,
             description: 'The characters to remove, defaults to whitespace'
         }
-    },
-    validate_arguments({ chars } = {}) {
-        if (isNotNil(chars) && !isString(chars)) {
```
Comment (jsnoble) [inline packages/data-mate/src/function-configs/string/trim.ts:110]: why was the argument validation taken out of this function?

File: packages/data-mate/src/function-configs/string/trimEnd.ts [hunk 2]
```diff
@@ -68,10 +68,5 @@ export const trimEndConfig: FieldTransformConfig<TrimEndArgs> = {
             array: false,
             description: 'The characters to remove, defaults to whitespace'
         }
-    },
-    validate_arguments({ chars } = {}) {
-        if (isNotNil(chars) && !isString(chars)) {
```
Comment (jsnoble) [inline packages/data-mate/src/function-configs/string/trimEnd.ts:73]: why was the argument validation taken from this function

File: packages/data-mate/src/function-configs/string/trimStart.ts [hunk 3]
```diff
@@ -62,10 +62,5 @@ export const trimStartConfig: FieldTransformConfig<TrimStartArgs> = {
             array: false,
             description: 'The characters to remove, defaults to whitespace'
         }
-    },
-    validate_arguments({ chars } = {}) {
```
Comment (jsnoble) [inline packages/data-mate/src/function-configs/string/trimStart.ts:66]: why was the argument validation taken from this function?

File: packages/data-mate/src/function-configs/date/toDate.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    argument_schema: {
+        format: {
+            type: FieldType.String,
+            description: `When the value is a string, this indicates the date string format.
+See https://date-fns.org/v2.16.1/docs/parse for more info.
+Default: iso_8601 for strings and epoch_millis for number`
+        },
+        time_resolution: {
+            type: FieldType.String,
+            description: `This will be set on the field to indicate whether the input date is stored in with millisecond or second accuracy.
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/date/toDate.ts:105]: typo - `in with`

File: packages/utils/src/strings.ts [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
-import {
-    trim as lTrim,
-    trimEnd as lTrimEnd,
-    trimStart as lTrimStart
-} from 'lodash';
+import lTrim from 'lodash/trim';
+import lTrimStart from 'lodash/trimStart';
+import lTrimEnd from 'lodash/trimEnd';
+import lCamelCase from 'lodash/camelCase';
+import lSnakeCase from 'lodash/snakeCase';
```
Comment (jsnoble) [inline packages/utils/src/strings.ts:6]: Is this ok to use? I thought there was an inherent problem how they split up words which is why we had our own, and styled our own. I am not sure where we ran into that problem, maybe they fixed it?

File: packages/data-mate/src/function-configs/date/toISO8601.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    accepts: [
+        FieldType.String,
+        FieldType.Number,
+        FieldType.Date
+    ],
+    argument_schema: {
+        time_resolution: {
+            type: FieldType.String,
+            description: `This will be set on the field to indicate whether the input date is stored in with millisecond or second accuracy.
```
Comment (ciorg) [inline packages/data-mate/src/function-configs/date/toISO8601.ts:63]: `stored in with` typo

**Review by jsnoble — CHANGES_REQUESTED** [review]

File: packages/data-mate/src/function-configs/string/trimEnd.ts (hunk 2 above)
Comment (peterdemartini) [reply packages/data-mate/src/function-configs/string/trimEnd.ts:73]: Same answer as above

File: packages/data-mate/src/function-configs/string/trimStart.ts (hunk 3 above)
Comment (peterdemartini) [reply packages/data-mate/src/function-configs/string/trimStart.ts:66]: Same answer as above

File: packages/utils/src/strings.ts (hunk 5 above)
Comment (peterdemartini) [reply packages/utils/src/strings.ts:6]: Potentially, but ours was currently wrong, so I decided to trust another implementation.

File: packages/data-mate/src/function-configs/date/toISO8601.ts (hunk 6 above)
Comment (peterdemartini) [reply packages/data-mate/src/function-configs/date/toISO8601.ts]: This should be fixed

File: packages/data-mate/src/function-configs/date/subtractFromDate.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+            );
+        };
+    },
+    accepts: [
+        FieldType.Date
+    ],
+    argument_schema: {
+        expr: {
+            type: FieldType.String,
+            description: `The date math expression used from calculate the date from subtract from.
```
[the remark this answers is not in the document]
Comment (peterdemartini) [reply packages/data-mate/src/function-configs/date/subtractFromDate.ts]: This should be fixed

File: packages/data-mate/src/function-configs/date/toDate.ts (hunk 4 above)
Comment (peterdemartini) [reply packages/data-mate/src/function-configs/date/toDate.ts]: This should be fixed

**Review by ciorg — APPROVED** [review]
LGTM

**Review by jsnoble — APPROVED** [review]

**Pull request merged**